util/estargz/labels.go

```diff
@@ -1,38 +1,84 @@
 package estargz
 
 import (
-	"fmt"
 	"strings"
 
-	ctdlabels "github.com/containerd/containerd/v2/pkg/labels"
-	"github.com/containerd/stargz-snapshotter/estargz"
+	"github.com/containerd/containerd/v2/pkg/labels"
 	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+const (
+	// TargetRefLabel is a label which contains image reference.
+	//
+	// It is a copy of [stargz-snapshotter/fs/source.targetRefLabel].
+	//
+	// [stargz-snapshotter/fs/source.targetRefLabel]: https://github.com/containerd/stargz-snapshotter/blob/v0.16.3/fs/source/source.go#L64-L65
+	TargetRefLabel = "containerd.io/snapshot/remote/stargz.reference"
+
+	// TargetDigestLabel is a label which contains layer digest.
+	//
+	// It is a copy of [stargz-snapshotter/fs/source.targetDigestLabel].
+	//
+	// [stargz-snapshotter/fs/source.targetDigestLabel]: https://github.com/containerd/stargz-snapshotter/blob/v0.16.3/fs/source/source.go#L67-L68
+	TargetDigestLabel = "containerd.io/snapshot/remote/stargz.digest"
+
+	// TargetImageLayersLabel is a label which contains layer digests contained in
+	// the target image.
+	//
+	// It is a copy of [stargz-snapshotter/fs/source.targetImageLayersLabel].
+	//
+	// [stargz-snapshotter/fs/source.targetImageLayersLabel]: https://github.com/containerd/stargz-snapshotter/blob/v0.16.3/fs/source/source.go#L70-L72
+	TargetImageLayersLabel = "containerd.io/snapshot/remote/stargz.layers"
+
+	// TargetSessionLabel is a label which contains session IDs usable for
+	// authenticating the target snapshot.
+	//
+	// It has no equivalent in github.com/containerd/stargz-snapshotter.
+	TargetSessionLabel = "containerd.io/snapshot/remote/stargz.session"
+)
+
+const (
+	// TOCJSONDigestAnnotation is an annotation for an image layer. This stores the
+	// digest of the TOC JSON.
+	// This annotation is valid only when it is specified in `.[]layers.annotations`
+	// of an image manifest.
+	//
+	// This is a copy of [estargz.TOCJSONDigestAnnotation]
```

Comment:
Why isn't this imported (or upstream constant just used instead). Same for next.

Reply:
I went back-and-forth a bit; initially I had these as alias (`TOCJSONDigestAnnotation = estargz.TOCJSONDigestAnnotation`), but as we already would have to define the non-exported ones locally (and the  `estargz` package containing much more than just the consts), and these consts to not change, I thought it would be OK to just have our own copy, as long as we refer to where they originate from.